Editor/Processors/SkinnedMeshes/MergeSkinnedMeshProcessor.cs

```diff
@@ -118,6 +120,36 @@ public override void Process(OptimizerSession session, MeshInfo2 target)
                 target.AssertInvariantContract($"processing meshInfo {Target.gameObject.name}");
             }
 
+#if !UNITY_2021_2_OR_NEWER
+            // material slot #4 should not be animated to avoid Unity bug
+            // https://issuetracker.unity3d.com/issues/material-is-applied-to-two-slots-when-applying-material-to-a-single-slot-while-recording-animation
+            const int SubMeshIndexToShiftIfAnimated = 4;
+            if (CheckAnimateSubMeshIndex(session, meshInfos, subMeshIndexMap, SubMeshIndexToShiftIfAnimated))
+            {
+                target.SubMeshes.Insert(SubMeshIndexToShiftIfAnimated, new SubMesh());
+
+                for (var i = 0; i < meshInfos.Length; i++)
+                {
+                    var meshInfo = meshInfos[i];
+                    mappings.Clear();
+
+                    for (var j = 0; j < meshInfo.SubMeshes.Count; j++)
+                    {
+                        var targetSubMeshIndex = subMeshIndexMap[i][j];
+                        if (targetSubMeshIndex >= SubMeshIndexToShiftIfAnimated)
+                        {
+                            mappings.Add(($"m_Materials.Array.data[{targetSubMeshIndex}]",
+                               $"m_Materials.Array.data[{targetSubMeshIndex + 1}]"));
+                        }
+                    }
```

Comment:
meshInfosをforeachする必要ないと思います。イカの形で多分十分化とお思います。

```suggestion
                mappings.Clear();
                for (var i = 4; i < target.SubMeshes.Length; i++)
                {
                    mappings.Add(($"m_Materials.Array.data[{i}]",
                               $"m_Materials.Array.data[{i + 1}]"));                    
                }
```

Reply:
対応しました。
二段階の move property は各 meshInfo ではなく target に対してやればよいということだったのですね。
元の Renderer を見る必要があるアニメーションのチェックだけ Destroy する前に残しています。